python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -0,0 +1,471 @@
+    meta : A GeoMeta object (optional)
+
+    Notes
+    -----
+    The GeoColumn class subclasses `NumericalColumn`. Combined with
+    `_copy_type_metadata`, this assures support for existing cudf algorithms.
+    """
+
+    def __init__(
+        self,
+        data: GeoArrowBuffers,
+        meta: GeoMeta = None,
+        shuffle_order: cudf.Index = None,
+    ):
+        base = cudf.Series(cudf.RangeIndex(0, len(data)))._column.data
+        super().__init__(base, dtype="int64")
+        self._geo = data
+        if meta is not None:
+            self._meta = meta
+        else:
+            self._meta = GeoMeta(data)
+        if shuffle_order is not None:
+            self._data = shuffle_order
+
+    def __iter__(self):
+        self._iter_index = 0
+        return self
+
+    def __next__(self):
+        if self._iter_index >= len(self):
+            raise StopIteration
+        result = self[self._iter_index]
+        self._iter_index = self._iter_index + 1
+        return result
+
+    def to_host(self):
+        result = GeoColumn(self._geo.to_host(), self._meta.copy(), self.data)
+        return result
+
+    def __getitem__(self, item):
+        """
+        Returns GpuGeometry objects for each of the rows specified by index.
+        """
+        if not isinstance(item, numbers.Integral):
+            raise NotImplementedError
+        return self.iloc[item]
+
+    @property
+    def loc(self):
+        """
+        Not currently supported.
+        """
+        return GeoColumnLocIndexer(self)
+
+    @property
+    def iloc(self):
+        """
+        Return the i-th row of the GeoSeries.
+        """
+        return GeoColumnILocIndexer(self)
+
+    def __len__(self):
+        """
+        Returns the number of unique geometries stored in this GeoColumn.
+        """
+        return len(self._meta.input_types)
+
+    @property
+    def points(self):
+        return self._geo._points
+
+    @property
+    def multipoints(self):
+        return self._geo._multipoints
+
+    @property
+    def lines(self):
+        return self._geo._lines
+
+    @property
+    def polygons(self):
+        return self._geo._polygons
+
+    def _dump(self):
+        return (
+            f"POINTS\n"
+            f"{self.points._repr__()}\n"
+            f"MULTIPOINTS\n"
+            f"{self.multipoints._repr__()}\n"
+            f"LINES\n"
+            f"{self.lines._repr__()}\n"
+            f"POLYGONS\n"
+            f"{self.polygons._repr__()}\n"
+        )
+
+    def __repr__(self):
+        return (
+            f"GeoColumn\n"
+            f"{len(self.points)} POINTS\n"
+            f"{len(self.multipoints)} MULTIPOINTS\n"
+            f"{len(self.lines)} LINES\n"
+            f"{len(self.polygons)} POLYGONS\n"
+        )
+
+    def copy(self, deep=True):
+        """
+        Create a copy of all of the GPU-backed data structures in this
+        GeoColumn.
+        """
+        result = GeoColumn(self._geo.copy(), self._meta.copy(), self.data)
```

Comment:
Should the deep parameter be passed to these copies? Should `self.data` get copied?

Reply:
Ashwin is creating a PR for cudf to add `Buffer.copy()`. I'm going to toss it into cudf for now.